cpp/src/shuffler/postbox.cpp

```diff
@@ -80,5 +95,8 @@ std::string PostBox::str() const {
     return ss.str();
 }
 
+// Explicit instantiation for PartID and Rank
+template class PostBox<PartID>;
+template class PostBox<Rank>;
```

Comment:
*nit*: This split of the template class into definition in header and implementation in cpp file, along with this explicit instantiation, means that to use a hypothetical newly templated `PostBox` we must remember to explicitly instantiate the template type _somewhere_. 

This is in contrast to the usual case where the definition of the template class in the header also contains the implementation.

Is this a deliberate choice?

Reply:
I have no strong opinion,  but I think the implementation is small/simple enough to include everything in `postbox.hpp`.

Reply:
@wence- It was a deliberate choice TBH. I wanted to make sure that we know explicitly how this PostBox templates are being used. In the impl cpp, we know the will be a PostBox with partition ID as the key as well as Rank as the key.